File: src/CSM.Domain/Entities/Finance/Vendor.cs
using CSM.Domain.Common;
using CSM.Domain.Entities.Procurement;
namespace CSM.Domain.Entities.Finance;

public class Vendor : TenantEntity
{
    public string VendorCode { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? ContactPerson { get; set; }

    public string? PhoneNumber { get; set; }

    public string? Email { get; set; }

    public string? Address { get; set; }

    public string? TaxIdentificationNumber { get; set; }

    public string? RegistrationNumber { get; set; }

    public string? BankName { get; set; }

    public string? BankAccountNumber { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<PurchaseOrder> PurchaseOrders { get; set; }
    = new List<PurchaseOrder>();

public ICollection<Equipment> Equipment { get; set; }
    = new List<Equipment>();

    public ICollection<Expense> Expenses { get; set; }
        = new List<Expense>();

    public ICollection<Invoice> Invoices { get; set; }
        = new List<Invoice>();
}

File: mobile/csm_app/lib/features/admin/procurement/data/models/vendor.dart
class Vendor {
  const Vendor({
    required this.id,
    required this.companyId,
    required this.vendorCode,
    required this.name,
    this.contactPerson,
    this.phoneNumber,
    this.email,
    this.taxIdentificationNumber,
    this.registrationNumber,
    this.bankName,
    this.bankAccountNumber,
    this.address,
    required this.isActive,
    required this.createdAtUtc,
    this.updatedAtUtc,
  });

  final String id;
  final String companyId;
  final String vendorCode;
  final String name;
  final String? contactPerson;
  final String? phoneNumber;
  final String? email;
  final String? taxIdentificationNumber;
  final String? registrationNumber;
  final String? bankName;
  final String? bankAccountNumber;
  final String? address;
  final bool isActive;
  final DateTime createdAtUtc;
  final DateTime? updatedAtUtc;

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['id'] as String,
      companyId: json['companyId'] as String,
      vendorCode: json['vendorCode'] as String? ?? '',
      name: json['name'] as String? ?? '',
      contactPerson:
          json['contactPerson'] as String?,
      phoneNumber:
          json['phoneNumber'] as String?,
      email: json['email'] as String?,
      taxIdentificationNumber:
          json['taxIdentificationNumber'] as String?,
      registrationNumber:
          json['registrationNumber'] as String?,
      bankName: json['bankName'] as String?,
      bankAccountNumber:
          json['bankAccountNumber'] as String?,
      address: json['address'] as String?,
      isActive: json['isActive'] as bool? ?? true,
      createdAtUtc:
          DateTime.parse(json['createdAtUtc'] as String),
      updatedAtUtc:
          json['updatedAtUtc'] == null
              ? null
              : DateTime.parse(
                  json['updatedAtUtc'] as String,
                ),
    );
  }
}
